packages/parse-mapping-lookup/src/ast.ts

```diff
@@ -0,0 +1,111 @@
+import debugModule from "debug";
+const debug = debugModule("parse-mapping-lookup:ast");
+
+/*
+ * Expression
+ */
+
+export interface Expression {
+  kind: "expression";
+  root: Identifier;
+  pointer: Pointer;
+}
+
+export const expression = (options: {
+  root: Identifier;
+  pointer: Pointer;
+}): Expression => {
+  const { root, pointer } = options;
+  return {
+    kind: "expression",
+    root,
+    pointer
+  };
+};
+
+/*
+ * Identifier
+ */
+
+export interface Identifier {
+  kind: "identifier";
+  name: string;
+}
+
+export const identifier = (options: { name: string }): Identifier => {
+  const { name } = options;
+  return {
+    kind: "identifier",
+    name
+  };
+};
+
+/*
+ * Pointer
+ */
+
+export interface Pointer {
+  kind: "pointer";
+  path: Step[];
+}
+
+export type Step = Access | Lookup;
+
+export const pointer = (options: { path: Step[] }): Pointer => {
+  const { path } = options;
+  return {
+    kind: "pointer",
+    path
+  };
+};
+
+/*
+ * Lookup
+ */
+
+export interface Lookup {
```

Comment:
These names -- `Lookup` vs `Access` -- are not very intuitive.  Can we replace them with something more explanatory?  Say `MemberAccess` and `IndexAccess`, matching Solidity?

Reply:
Updated to `MemberLookup` and `IndexAccess` if you're good with that.

Reply:
Sure, that's sufficiently unambiguous.